### PR TITLE
Nevra + Patch advisory name filter - first implementation of UI

### PR DESCRIPTION
**Pull request by lneves12**
## What does this PR change?

Nevra + Patch name filter - first implementation of UI

## GUI diff

Screenshot from 2019-05-09 00-40-34
Screenshot from 2019-05-09 00-40-19

- [x] **DONE**

## Documentation
- No documentation needed.

- [ ] **DONE**

## Test coverage
- No tests.

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"  
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests" 		 
- [ ] Re-run test "ruby_rubocop"
- [x] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql" (Test skipped, there are no changes to test)		 
- [ ] Re-run test "susemanager_unittests" (Test skipped, there are no changes to test)
- [ ] Re-run test "javascript_lint" (Test skipped, there are no changes to test)

### CODE REVIEW

**Comment by lneves12** [comment]
@hustodemon @ncounter this is still work in progress, but some early feedback and testing would be welcome.

**Comment by hustodemon** [comment]
Just to sum up implemented  filters in backend (entity - matcher - field):
- package - contains - name (implemented in this PR)
- package - equals - nevra (implemented in this PR)
- package - equals - nevr (implemented in this PR (when user doesn't fill the `architecture` field))
- patch - equals - advisory_name - **not** implemented in this PR (i think equals makes more sense here than contains)

I tried the first three cases and they were working great! The 4th one (patch) should be adjusted (it also depends on my PR  - https://github.com/uyuni-project/uyuni/pull/932, which is yet to be merged).

**Comment by lneves12** [comment]
@hustodemon Thanks for the early testing.

I rebased the branch on top of yours and adapted the code to the new filter: patch - equals - advisory_name

The success case should be working for everything, I need to clean up some code and implement better form error handling

**Comment by lneves12** [comment]
@hustodemon the PR is ready for review. It should be easily reviewable commit by commit

File: java/code/src/com/suse/manager/webui/controllers/contentmanagement/handlers/ProjectHandler.java [hunk 1]
```diff
@@ -97,6 +103,10 @@ public static NewProjectRequest getProjectRequest(Request req) {
             requestErrors.put("name", "Name must not exceed 128 characters");
         }
 
+        ContentManager.lookupProjectByNameAndOrg(projPropsRequest.getName(), user).ifPresent(cp -> {
```
[the remark this answers is not in the document]
Comment (lneves12) [reply java/code/src/com/suse/manager/webui/controllers/contentmanagement/handlers/ProjectHandler.java]: Ah damn, well caught, I broke the properties update after adding this validation. Fixed this in my last commit and also the description of the options NEVRA.

I am currently investigating a way of adding a confirmation modal on the filters+environments deletion

File: java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProject.java [hunk 2]
```diff
@@ -331,7 +331,7 @@ public void attachFilter(ContentFilter filter) {
         if (idx != -1) {
             ContentProjectFilter toUpdate = filters.get(idx);
             if (toUpdate.getState() == ContentProjectFilter.State.DETACHED) {
-                toUpdate.setState(ContentProjectFilter.State.BUILT);
+                toUpdate.setState(ContentProjectFilter.State.EDITED);
```
Comment (hustodemon) [inline java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProject.java]: This is not entirely correct. What if the filter wasn't edited at all? OTOH if we wanted to do this properly, we'd need to track the previous state of the object :-/

File: java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProjectFactory.java [hunk 3]
```diff
@@ -163,7 +163,7 @@ private static void remove(ContentEnvironment contentEnvironment) {
      * @return matching Environment
      */
     public static Optional<ContentEnvironment> lookupEnvironmentByLabelAndProject(String label,
-            ContentProject project) {
+                                                                                  ContentProject project) {
```
Comment (hustodemon) [inline java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProjectFactory.java]: Please avoid changing the format.

File: java/code/src/com/suse/manager/webui/controllers/contentmanagement/handlers/ProjectHandler.java [hunk 4]
```diff
@@ -103,9 +104,11 @@ public static NewProjectRequest getProjectRequest(Request req) {
             requestErrors.put("name", "Name must not exceed 128 characters");
         }
 
-        ContentManager.lookupProjectByNameAndOrg(projPropsRequest.getName(), user).ifPresent(cp -> {
-            requestErrors.put("name", "Name already exists");
-        });
+        if (!editing) {
```
[the remark this answers is not in the document]
Comment (lneves12) [reply java/code/src/com/suse/manager/webui/controllers/contentmanagement/handlers/ProjectHandler.java]: yeah, you are right, it's not entirely correct yet. I changed the logic here, I removed the editing logic for a more generic one.

File: java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProject.java (hunk 2 above)
Comment (lneves12) [reply java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProject.java]: I extracted this logic to a new PR, let's discuss it there.

https://github.com/uyuni-project/uyuni/pull/966

**Pull request merged**